runtime-modules/content-directory/src/lib.rs

```diff
@@ -0,0 +1,1642 @@
+                class.schemas.push(schema);
+            });
+
+            Ok(())
+        }
+
+        pub fn update_class_schema_status(
+            origin,
+            class_id: T::ClassId,
+            schema_id: SchemaId,
+            schema_status: bool
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_known_class_id(class_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Check that schema_id is a valid index of class schemas vector:
+            Self::class_by_id(class_id).ensure_schema_id_exists(schema_id)?;
+            <ClassById<T>>::mutate(class_id, |class| {
+                class.update_schema_status(schema_id, schema_status)
+            });
+            Ok(())
+        }
+
+        pub fn set_class_property_lock_status_at_index(
+            origin,
+            class_id: T::ClassId,
+            in_class_schema_property_id: PropertyId,
+            is_locked: PropertyLockingPolicy
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_known_class_id(class_id)?;
+
+            // Ensure property_id is a valid index of class properties vector:
+            Self::class_by_id(class_id).ensure_property_id_exists(in_class_schema_property_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <ClassById<T>>::mutate(class_id, |class| {
+                class.set_property_lock_status_at_index(in_class_schema_property_id, is_locked)
+            });
+            Ok(())
+        }
+
+        pub fn set_reference_property_same_controller_status(
+            origin,
+            class_id: T::ClassId,
+            in_class_schema_property_id: PropertyId,
+            same_controller: SameController
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_known_class_id(class_id)?;
+
+            // Ensure property_id is a valid index of class properties vector:
+            Self::class_by_id(class_id).ensure_property_id_exists(in_class_schema_property_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            <ClassById<T>>::mutate(class_id, |class| {
+                class.set_reference_property_same_controller_status(in_class_schema_property_id, same_controller)
+            });
+            Ok(())
+        }
+
+        /// Update entity permissions.
+        ///
+
+        pub fn update_entity_permissions(
+            origin,
+            entity_id: T::EntityId,
+            controller: Option<EntityController<T>>,
+            frozen_for_controller: Option<bool>,
+            referenceable: Option<bool>
+        ) -> dispatch::Result {
+            perform_lead_auth::<T>(origin)?;
+
+            Self::ensure_known_entity_id(entity_id)?;
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            if let Some(controller) = controller {
```

Comment:
Done https://github.com/Joystream/joystream/pull/470/commits/7d13d180cc68482de632378ff07bd621bf023a97

Reply:
This was not exactly what I meant, because now have introduced a mutable variable over even larger lexical scope. Compared to the issue of writing multiple times, I would say this is perhaps even less ideal. However, for now, I think this is not worth spending more time on.